scripts/nbconverted/explore_relationship_AandB_pt2.py

```diff
@@ -0,0 +1,682 @@
+        
+        weights_decoder_file: file containing the learned weights associated with the vae decoder model
+        
+        encoded_dir:  directory to use to output offset vector to 
+
+        gene_id: gene you are using as the "phenotype" to sort samples by 
+
+                 This gene is referred to as "target_gene" in comments below
+
+
+        percent_low: integer between 0 and 1
+
+        percent_high: integer between 0 and 1
+        
+        out_dir: directory to output predicted gene expression to
+
+    computation:
+        1.  Sort samples based on the expression level of the target gene defined by the user
+        2.  Sample_data are encoded into VAE latent space
+        3.  We predict the expression profile of the OTHER genes at a given level of target gene 
+            expression by adding a scale factor of offset vector to the sample
+
+            The scale factor depends on the distance along the target gene expression gradient
+            the sample is.  For example the range along the target gene expression is from 0 to 1.  
+            If the sample of interest has a target gene expression of 0.3 then our prediction
+            for the gene expression of all other genes is equal to the gene expression corresponding
+            to the target gene expression=0 + 0.3*offset latent vector
+        3.  Prediction is decoded back into gene space
+        4.  This computation is repeated for all samples 
+
+    output: 
+         1. encoded predicted expression profile per sample
+         2. predicted expression profile per sample
+
+    """
+
+    # Load arguments
+    offset_file = os.path.join(encoded_dir, "offset_latent_space_vae.txt")
+
+    # Output file
+    predict_file = os.path.join(out_dir, "shifted_predicted_gene_exp.txt")
+    predict_encoded_file = os.path.join(out_dir, "shifted_predicted_encoded_gene_exp.txt")
+
+    # Read in data
+    target_gene_data = all_data[gene_id]
+    offset_encoded = pd.read_table(offset_file, header=0, index_col=0)    
+    
+    # read in saved VAE models
+    loaded_model = load_model(model_encoder_file)
+    loaded_decoder_model = load_model(model_decoder_file)
+
+    # load weights into models
+    loaded_model.load_weights(weights_encoder_file)
+    loaded_decoder_model.load_weights(weights_decoder_file)
+    
+    # Initialize dataframe for predicted expression of sampled data
+    predicted_sample_data = pd.DataFrame(columns=sample_data.columns)
+    predicted_encoded_sample_data = pd.DataFrame()
+    
+    sample_ids = sample_data.index
+    for sample_id in sample_ids:
+        sample_exp = sample_data.loc[sample_id].to_frame().T
+        
+        # Use trained model to encode expression data into SAME latent space
+        predict = loaded_model.predict_on_batch(sample_exp)
+
+        predict_encoded_df = pd.DataFrame(predict)
+        predicted_encoded_sample_data = predicted_encoded_sample_data.append(predict_encoded_df, ignore_index=True)
```

Comment:
Up to you, but I highly recommend changing lines like this into:
```python
predicted_encoded_sample_data = (
  predicted_encoded_sample_data
  .append(predict_encoded_df, ignore_index=True)
)
```
This works for any line that has multiple or single functions being called.